scripts/cli/src/util/sqlite3.ts

```diff
@@ -0,0 +1,40 @@
+import sql from "sqlite3";
+import type { KeyringPair } from "@polkadot/keyring/types";
+
+const sql3 = sql.verbose();
+// Setting up a database for storing data.
+const db = new sql3.Database("database.db");
+
+export function createTable() {
+  db.run(
+    "CREATE TABLE IF NOT EXISTS next_nonce ( account TEXT PRIMARY KEY, nonce INT DEFAULT 0 )"
+  );
+}
+
+export function insertNonce(accountAddr: string) {
+  db.run("INSERT OR REPLACE INTO next_nonce(account) VALUES($account)", {
+    $account: accountAddr,
+  });
+}
+
+export function incrementNonce(accountAddr: string) {
+  db.run(
+    "INSERT INTO next_nonce(account) VALUES($account) ON CONFLICT(account) DO UPDATE SET nonce=nonce+1",
+    { $account: accountAddr }
+  );
+}
+
+export async function getNonce(signer: KeyringPair) {
+  return new Promise<number>((resolve, reject) => {
+    db.get(
+      "SELECT account, nonce FROM next_nonce WHERE account = $account",
+      { $account: signer.address },
+      (err, row) => {
+        if (err) reject(err);
+
+        console.log(`accound: ${row.account} nonce: ${row.nonce}`);
+        resolve(row.nonce);
+      }
+    );
+  });
+}
```

Comment:
The `geNonce` and `incrementNonce` need to be one atomic SQL statement.  Otherwise there will be a race between multiple processes.

The SQL in this `getNonce` version will return the current `nonce` and also increment the stored `nonce` for the `$account`:
```suggestion
export async function getNonce(signer: KeyringPair) {
  return new Promise<number>((resolve, reject) => {
    db.get(
      "INSERT INTO next_nonce(account) VALUES($account) ON CONFLICT(account) DO UPDATE SET nonce=nonce+1 RETURNING nonce;",
      { $account: signer.address },
      (err, row) => {
        if (err) reject(err);

        console.log(`accound: ${signer.account} nonce: ${row.nonce}`);
        resolve(row.nonce);
      }
    );
  });
}
```

Reply:
Also don't need the `insertNonce` function since the `getNonce` will insert a zero if the account wasn't used before.